Python/binary_exponentiation.py

```diff
@@ -0,0 +1,17 @@
+#To compute a ^ b, this code runs in O(log b) instead of O(b)
+
+def BinaryExpo(a, b):
+    if b == 0:
+        return(1)
+    res = BinaryExpo(a, b // 2)
+    if b % 2 == 1:
+        return(res * res * a)
+    else:
+        return(res * res)
+
+#driver code
+a, b = [int(x) for x in input().split()]
+print(BinaryExpo(a, b))
+
+#input: 2 3
```

Comment:
Do not use SINGLE LINE COMMENTS.
Add Sample I/P, O/P, Time and Space complexity in a single MULTILINE COMMENT.
Also, please create an issue first.
Then a PR.

Do all of this and tag me on the issue and here as well.
I'll merge it. 👍